sdks/python/apache_beam/runners/dataflow/dataflow_runner.py

```diff
@@ -415,6 +416,12 @@ def run_pipeline(self, pipeline, options, pipeline_proto=None):
       self.proto_pipeline, self.proto_context = pipeline.to_runner_api(
           return_context=True, default_environment=self._default_environment)
 
+    if any(pcoll.is_bounded == beam_runner_api_pb2.IsBounded.UNBOUNDED
+           for pcoll in self.proto_pipeline.components.pcollections.values()):
+      options.view_as(StandardOptions).streaming = True
```

Comment:
if users specify streaming, can we honor the one specified by the users? This at least can let users keep the current behavior.